Get Sign-In and Sign-Up working when running FOAM itself.

jlhughes opened the pull request:
# FOAM Sign-In and Sign-up
Get Sign-In and Sign-Up working when running FOAM itself. 

#### admin login 
username: foam-admin
password: foam-admin

## Sign-In 
Sign-In working with VirtualHostRoutingServlet. 

## Sign-up 
- Sign-up creates users

## Password management
In addition to password changes secured via tokens or email codes, a less secure direct update system is provided for simple/small FOAM applications which don't use or have access to an email server.  
See notes in src/groupPermissionJunctions.jrl to grant the appropriate permissions to basicUser. 

## Other FOAM issues and quirks 
**appears to be a u3 issue**
- delete action missing
- some dao tableview fail to render models, but count is correct. 
- some canned queries are not selectable

noodlemoodle committed:
signin signup loginview v2 and clientloginservice initial model

noodlemoodle committed:
switch to nanos/auth/login/loginview

noodlemoodle committed:
fix action labels and actionview width

noodlemoodle committed:
merge dev

noodlemoodle committed:
adding wizard flow

noodlemoodle committed:
remove loginfailed

noodlemoodle committed:
Merge branch 'development' of https://github.com/kgrgreer/foam3 into clientLoginService

noodlemoodle committed:
Merge branch 'development' of https://github.com/kgrgreer/foam3 into clientLoginService

noodlemoodle committed:
add wizardrunner fix

noodlemoodle committed:
add showaction to resetpassword

noodlemoodle committed:
add fix for altflowsaver

noodlemoodle committed:
fix messages

noodlemoodle committed:
add reporting if analyticable

noodlemoodle committed:
fix retrievepassword for resetbycode

noodlemoodle committed:
add clientloginservice to services

noodlemoodle committed:
Merge branch 'development' of https://github.com/kgrgreer/foam3 into clientLoginService

noodlemoodle committed:
set trim true on identifier

noodlemoodle committed:
change loginview

noodlemoodle committed:
fix

noodlemoodle committed:
add padding

noodlemoodle committed:
add property availability checks to signup

noodlemoodle committed:
use validationpredicates

noodlemoodle committed:
update validationpredicates

noodlemoodle committed:
fix validationpredicates

noodlemoodle committed:
fix validationPredicate

noodlemoodle committed:
Merge branch 'development' of https://github.com/kgrgreer/foam3 into clientLoginService

noodlemoodle committed:
use wizard emailverification

jlhughes committed:
Sign-In working with VirtualHostRoutingServlet. BaseUnAuthBorder disabled for the moment

jlhughes committed:
re: BaseUnAuthBorder - only theme change was necessary to get things working

jlhughes committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into SignInSignUp

jlhughes committed:
Merge branch 'clientLoginService' of https://github.com/kgrgreer/foam3 into SignInSignUp

jlhughes committed:
resolve differences from clientloginservice

jlhughes committed:
Merge branch 'SignInSignUp' of github.com:kgrgreer/foam3 into SignInSignUp

jlhughes committed:
make duplicate email permissioned so only applies for apps that want it. foam itself doesn't care, only username needs to be unique.

jlhughes committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into SignInSignUp

noodlemoodle committed:
fix double loginview

jlhughes committed:
Merge branch 'SignInSignUp' of github.com:kgrgreer/foam3 into SignInSignUp

noodlemoodle committed:
update default lifecyclestate

noodlemoodle commented:
TODO: merge [this](https://github.com/nanoPayinc/NANOPAY/pull/18248) when needed

noodlemoodle committed:
fix signin signup on enter

noodlemoodle committed:
signup username validation partial fix

jlhughes committed:
default LifecycleState to ACTIVE

jlhughes committed:
moved from nanopay

jlhughes committed:
move basicUser from nanopay

jlhughes committed:
Move token npecs from nanopay to foam

jlhughes committed:
Merge branch 'SignInSignUp' of github.com:kgrgreer/foam3 into SignInSignUp

jlhughes committed:
update documentation

jlhughes committed:
remove test for PENDING now that foam User defaults LifecycleState to ACTIVE

jlhughes committed:
User - remove validation check for verified email - applications can impose this is they desire

jlhughes committed:
editor cleanup up javascript ;

jlhughes committed:
restrict unique check to not-deleted

jlhughes committed:
additional moves from nanopay

jlhughes committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into SignInSignUp

jlhughes committed:
debugging support to output which nspec skeleton is serving

jlhughes committed:
permissions for notification settings

noodlemoodle committed:
add capabilityauthservice to foam auth

noodlemoodle committed:
add ucj and a group permission

noodlemoodle committed:
fix username availability/validity check

noodlemoodle committed:
add enabled check for signin signup actions

jlhughes committed:
revert username available to boolean

jlhughes committed:
change explicit view setup to pushMenu

jlhughes committed:
ensure system X is used for dao queries for existing users

jlhughes committed:
anonymous permissions to own group, foam spid, and userPropertyAvailabilityService for user sign-up

jlhughes committed:
handle existing invalid condition

jlhughes committed:
Merge branch 'development' of https://github.com/kgrgreer/foam3 into SignInSignUp

jlhughes committed:
add fileDAO

jlhughes committed:
ResetPasswordInsecure - allows direct password update - without email, tokens

Meant for small foam applications which do not use or have access to email servers.
Enabled via permissions:
service.resetPasswordInsecureService
service.userUpdatePasswordDAO
menu.read.reset-password

jlhughes committed:
update documentation

jlhughes committed:
Merge branch 'SignInSignUp' of https://github.com/kgrgreer/foam3 into SignInSignUp

kgrgreer commented on src/foam/nanos/auth/login/SignUp.js:
```diff
@@ -0,0 +1,131 @@
+    {
+      class: 'String',
+      name: 'username',
+      label: 'Username',
```
Redundant

kgrgreer commented on src/foam/nanos/auth/resetPassword/ResetPasswordInsecure.js:
```diff
@@ -0,0 +1,72 @@
+          this.window.history.replaceState(null, null, this.window.location.origin);
+          this.pushMenu("");
+
+          this.ctrl.add(this.NotificationMessage.create({
```
Using 'ctrl' directly is discouraged because then it directly couples us to the ApplicationController and prevents us from using new ones like ZAC. Instead of importing ctrl, can you import displayToastMessage and use it instead?

sarthak-marwaha replied:
import notify as that does a put to notification dao and then calls displayToastMessage itself.

sarthak-marwaha commented on src/foam/nanos/auth/login/LoginView.js:
```diff
@@ -0,0 +1,477 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
```
This was refactored a while ago and now is way smaller. So the new one should be copied over to here or this should be removed https://github.com/kgrgreer/foam3/blob/development/src/foam/u2/view/LoginView.js

sarthak-marwaha replied:
Looking at the code here there are a few changes that will need to be carried over. But there is still a lot of extras in this file that are no longer used/required

sarthak-marwaha commented on src/foam/nanos/auth/login/ClientLoginService.js:
```diff
@@ -0,0 +1,188 @@
+    {
+      name: 'notifyUser_',
+      code: function(err, msg, type) {
+        this.ctrl.add(this.NotificationMessage.create({
```
Should use notify that it imports

sarthak-marwaha commented on src/foam/nanos/auth/login/SignUp.js:
```diff
@@ -0,0 +1,131 @@
+        }
+      ]
+    },
+    {
```
Has been updated in https://github.com/kgrgreer/foam3/commit/0cb137d38897cf1ecc77db23eb9e0e842516fe38#diff-0026cfc7e2391117957f00f1cc83489c816b140fa9533d11f7b475b2cf7d7ad7

sarthak-marwaha commented on src/foam/nanos/controller/ApplicationController.js:
```diff
@@ -499,19 +510,25 @@ foam.CLASS({
-    function render() {
-      var self = this;
-      self.addMacroLayout();
+    async function render() {
```
Dont think this change to render is required or current, is this a merge carryover?

sarthak-marwaha commented on src/foam/nanos/controller/ApplicationController.js:
```diff
@@ -423,23 +436,21 @@ foam.CLASS({
+    async function onClientLoad() {
       let self = this;
-      this.clientPromise.then(async function(client) {
+      await this.clientPromise.then(async function(client) {
```
Should not be awaited. onClientLoad() should just be called and any changes due to it should happen in this then()

sarthak-marwaha commented on src/foam/nanos/theme/Theme.js:
```diff
@@ -206,8 +206,9 @@ foam.CLASS({
       name: 'loginView',
       documentation: `Border used to wrap unauthenticated views like loginViews. Set to nullBorder for no additional borders`,
-      value: 'foam.u2.borders.BaseUnAuthBorder'
+      value: 'foam.nanos.auth.login.LoginView'
```
This should be BaseUnAuthBorder as the loginView will always be the one LoginView we use across apps. This property should only be the border that is used around the loginView

sarthak-marwaha commented on src/foam/nanos/u2/navigation/SignUp.js:
```diff
@@ -126,11 +126,15 @@ foam.CLASS({
         {
-          args: ['emailAvailable', 'email'],
-          query: 'emailAvailable!="unavailable"',
+          args: ['email'],
```
Changes to this file should be reverted. This was updated https://github.com/kgrgreer/foam3/commit/0cb137d38897cf1ecc77db23eb9e0e842516fe38#diff-0026cfc7e2391117957f00f1cc83489c816b140fa9533d11f7b475b2cf7d7ad7

sarthak-marwaha commented on src/foam/nanos/u2/navigation/SignUp.js:
```diff
@@ -185,12 +188,20 @@ foam.CLASS({
-        if ( ! desiredPassword || desiredPassword.length < 10 ) return this.PASSWORD_ERR;
-        if ( ! passwordAvailable ) return this.WEAK_PASSWORD_ERR;
-      }
+      validationPredicates: [
```
This one can stay

sarthak-marwaha commented on src/foam/nanos/auth/login/ClientLoginService.js:
```diff
@@ -0,0 +1,188 @@
+      }
+    },
+    {
+      name: 'wizardVerifyEmail',
```
Not sure why we need both of these. We could always use the wizard version

sarthak-marwaha commented on src/foam/nanos/auth/login/LoginView.js:
```diff
@@ -0,0 +1,477 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
```
Relevant commits: 
https://github.com/kgrgreer/foam3/commit/9787a3656492088112cce7e7502b8074dd7ca2c9
https://github.com/kgrgreer/foam3/commit/22f10e7c9fcb71c11b0da511ebb0e80edfd7d1a1
https://github.com/kgrgreer/foam3/commit/87c508a2d9a8b3d331bcb5b128450ac50612631a
https://github.com/kgrgreer/foam3/commit/7077215d1e8adf02ee36054d15c50c37dae02443#diff-39db26af7503c6abf62d31fbd2bd5c2dc4b2ca1f47bc8c9822de87995d7cd972
https://github.com/kgrgreer/foam3/commit/564f57265c9b9b38a3e888712b4faf59232d8340#diff-39db26af7503c6abf62d31fbd2bd5c2dc4b2ca1f47bc8c9822de87995d7cd972
https://github.com/kgrgreer/foam3/commit/4db24ea808273be93047255700911330481d3cbf#diff-39db26af7503c6abf62d31fbd2bd5c2dc4b2ca1f47bc8c9822de87995d7cd972
https://github.com/kgrgreer/foam3/commit/2c9834d8c4a576b8c340826b3a459705b16c2f54

jlhughes committed:
SignInSignIpNotify changes (#3584)

* Replace explicit NotificationMessage.create with this.notify

Addressing PR comments
With the change to this.notify, no notification are displayed, as
notificationDAO is undefined in ApplicationController.
Also, with changes to SignIn, SignUp they fail to login, signup,
with no errors.

* passwordAvailable remains boolean

* correct case of SignUp username: userName -> username. This fixes the SignUp flow

* During SignIn and SignUp the anonymous user is used for notifications

and as a results requires permission to notificationDAO

* add fetchSubject back on request

jlhughes committed:
Merge branch 'SignInSignUp' of https://github.com/kgrgreer/foam3 into SignInSignUp

noodlemoodle committed:
update loginview

noodlemoodle committed:
fix resetpassword visibility

jlhughes committed:
system user requires a spid for system operations ocrring during replay

noodlemoodle committed:
add signInWithOIDC to clientloginservice

noodlemoodle committed:
use loginservice signinwithoidc

jlhughes committed:
read-add clien File decorators to user dao

jlhughes committed:
run.sh - only source env.sh if running locally - development build

jlhughes committed:
moving file entries together

jlhughes committed:
UseRegistrationSanitaiontDAO - re-add referral check

noodlemoodle committed:
fix  actionview

noodlemoodle committed:
fix login on enter

adamvy commented on src/foam/nanos/auth/PreventDuplicateEmailLoginService.js:
```diff
@@ -29,9 +29,15 @@ foam.CLASS({
+              EQ(User.EMAIL, identifier.toLowerCase()),
+              EQ(User.USER_NAME, identifier)
+            ),
+            EQ(User.LIFECYCLE_STATE, LifecycleState.ACTIVE)
```
This seems wrong.  We shouldn't let somebody create a user with the same email as some other deactivated user should we?

I feel like there should only ever be one user with a given email for a given spid, regardless of whether that user is active or inactive.

jlhughes replied:
@adamvy only username is unique.  unique email is enforced by the application by enabling the associated rules.  
username is only used for login, nothing internal depends on it. 
So after a user is 'deleted', in the base foam system, I don't see any reason not to allow reuse.  If an application wants another strategy then it can impose that.  
The only compromise here would be to have this check be a rule like duplicate email, and then it's easy for an application to enable/disable.

adamvy replied:
This doesn't check if a user is deleted though, it checks if they are ACTIVE.  Meaning we could have a PENDING user and an ACTIVE user with the same email.  But there isn't a unique email check on moving a user from PENDING to ACTIVE

adamvy replied:
Or in theory even many PENDING users all with the same email. Maybe that's desired? Just feels surprising I guess.

jlhughes replied:
Condition removed.  This DAO is used in nspec uniqueUserService which itself adds the condition LifecycleState != DELETED. And since this condition is in the nspec it makes it easy for an application to override.

adamvy commented on src/foam/nanos/auth/User.js:
```diff
@@ -645,8 +646,17 @@ foam.CLASS({
+      updateVisibility: 'RW',
+      readVisibility: 'RO',
+      documentation: `
+        Setting default lifecyclestate as ACTIVE and moving PENDING
```
This documentation reads more like a commit message that will lose its meaning in a few months.  Documentation here should describe what the lifecycleState property of a user is used for.

jlhughes replied:
Removed documentation.

adamvy commented on src/foam/nanos/auth/UserRegistrationDAO.js:
```diff
@@ -0,0 +1,122 @@
+          user.setGroup(theme.getRegistrationGroup());
+        }
+
+        // User duplicate = (User) getDelegate().find_(getX(),
```
Let's delete the commented out code if its not needed.

adamvy commented on src/foam/nanos/auth/UserRegistrationDAO.js:
```diff
@@ -0,0 +1,122 @@
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'UserRegistrationDAO',
+  extends: 'foam.dao.ProxyDAO',
```
Alternatively you could extend NullDAO to get the do-nothing impelmentation for most methods and then manually define a delegate property that you use for put().

But either way is probably fine.

adamvy commented on src/foam/nanos/auth/UserRegistrationDAO.js:
```diff
@@ -0,0 +1,122 @@
+      `
+    },
+    {
+      name: 'register',
```
would be nice to have documentation: describing what this method does

jlhughes replied:
doc added to the model.

adamvy commented on src/foam/nanos/auth/login/ClientLoginService.js:
```diff
@@ -0,0 +1,177 @@
+      code: async function(x, data, wizardFlow) {
+        var analyticable = foam.nanos.analytics.Analyticable.isInstance(data);
+        try {
+          ctrl.groupLoadingHandled = true;
```
This is referencing the global ctrl and not the import, should be `this.ctrl.groupLoadingHandled`

jlhughes replied:
corrected.

adamvy commented on src/users.jrl:
```diff
@@ -28,3 +26,24 @@ p({
+  lastName:"Foam",
+  email:"anonymous-foam@foam.foam",
+  emailVerified:true,
+  password:"WsJlIJDGGGQ=:cnE+tRNtF/yhOheF2POyNuocgKTZDVyXia7dITQFUEVPS9hpuQqp8lVTTm2iQQXbYk/8cBUCOV/prsUkO793Sg==",
```
should the anonymous user have a password?  Seems safer if we have no password and somehow disable password login for the anonymous user

adamvy replied:
bump

adamvy commented on src/menus.jrl:
```diff
@@ -321,7 +361,7 @@ p({
     "view":{
       class: "foam.u2.borders.BaseUnAuthBorder",
-      children: [ {"class":"foam.u2.view.LoginView", "mode_":"SignIn"} ]
+      children: [ {"class":"foam.nanos.auth.login.LoginView", "mode_":0} ]
```
It looks like you've completely replace the login view and associated signup/signin models.  Can we delete them LoginView and assocaited SignUp/SignIn models?

jlhughes replied:
in another PR.

adamvy commented on src/foam/nanos/auth/services.jrl:
```diff
@@ -459,3 +483,73 @@ p({
+  """
+})
+
+// only available to admin to update passwords.
```
The comment is wrong, this is also exposed to the basicUser group.

Overall this service feels unnecessary, can't we use property level permissions to let users change their own password (and let admin change anything?)

jlhughes replied:
Changing one's own password uses a DAO stack that has the PasswordHashingDAO. 
When admin, and changing another users password, there does not exist an nspec which to submit to to invoke the hashing of the desired password. 
if this is granted to basicUser then that is a mistake, ther service.daoname permission should not be granted to anyone.  
So open to ideas on how to allow administrators to change another users password.  The current change password views require either email or token verification, or the recent 'insecure' system which only allows change of the subject/context (already logged in users) password.

adamvy replied:
Here you expose service.userUpdatePasswordDAO to "basicUser"

https://github.com/kgrgreer/foam3/pull/3579/files/ce70ed0438e8fb343c4569ab2f67a82aa6de237f#diff-13acd8f684fcd591af01935a858aff229a504b34ee3ca83176f9118144228f46R41

adamvy replied:
this feels flaky though, can't we just decorate UserDAO in all cases to hash the password if the password field changes?  Weird that we have to expose the userDAO as these other special services to hash the passwords.

jlhughes replied:
Services and permissions fixed. 
We have pre-existing password protection services with email and token verification, which rely on verifying secondary information before allowing a password update.  These are onerous for a simple foam application, but also preclude placing the UserPasswordHashingDAO on userDAO.  The resetPasswordInsecureService was added for the simple app, with the only protection being that one can only update the logged in user password (your own).   Also, hitherto, it was not possible for admin to update another users password outside of using a script.  So the non-permissioned userUpdatePasswordDAO was added which allows both the user id and password to be provided. 
I agree that this seems a little much for password management, and I was surprised by lack of support the admin and the simple case.  I'll put more thought into a better system, and open to your ideas.  But for the moment this gets us status-quo.  The user only interacts via the Change Password menu, so we have lots of leeway to change the backend.

adamvy replied:
Sounds good, thanks for responding to my comments, I'm also still getting up to speed on how auth has evolved in nanos.

jlhughes replied:
Ok, previous to the clientLoginService PR, things worked more like you expected. 
I've reverted to that setup, allowing removal of nspec additions for userUpdatePassword for admin, and the PasswordResetInsecureService for users. 
An application has full control to change the userDAO stack, which nanopay does (for example). 
And if an application allows access to the User model by a User, then a user could update password directly from the desiredPassword field.

adamvy commented on src/foam/nanos/theme/themes.jrl:
```diff
@@ -3,11 +3,12 @@ p({
   "topNavigation":"foam.nanos.u2.navigation.ResponsiveTopNav",
   "defaultMenu": ["admin.data"],
-  "logo":"/images/foam_red.png",
+  "logo":"images/foam_red.png",
```
This breaks the logo for me when I hit

http://localhost:8080/foam3/src/foam/nanos/controller/index.html

running from raw foam3 nanos

jlhughes replied:
Not sure how to solve this one for both parties. 
When are running/using just javascript foam? Meaning - if I understand the use case, maybe it will illicit a solution.

jlhughes replied:
While working on this I can't find foam_red.svg or foam_red.png anywhere in the repository.

jlhughes replied:
In the original clientloginservice work this was not working, but not the original path is working for all cases.

adamvy replied:
I'm pretty sure /images/foam_red.png should work for all cases?  Doesn't the ImagesServlet serve all images from the file tree under /images/... ?

adamvy commented on src/foam/nanos/auth/login/SignIn.js:
```diff
@@ -0,0 +1,49 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
```
Tried running this locally, the LoginView seems really narrow?

<img width="528" alt="Screenshot 2024-11-23 at 12 21 38 PM" src="https://github.com/user-attachments/assets/5373c343-df03-42ad-af56-feb7767defcf">

jlhughes replied:
I've never seen this particular view/layout. 
What are your steps to reproduce?

adamvy replied:
looks like https://github.com/kgrgreer/foam3/pull/3585 re-does this view significantly, this is probably an irrelevant comment then

jlhughes replied:
Addressed.

adamvy commented on src/foam/nanos/auth/login/LoginView.js:
```diff
@@ -0,0 +1,477 @@
+    'appConfig',
+    'clientLoginService',
+    'loginVariables',
+    'memento',
```
I'm getting `debug.js:425 Assertion failed: Missing required import: memento in foam.nanos.auth.login.LoginView`

I don't think we have memento available at logged out time.  Also looks like we dont' actually use it so we should probably remove the import?

jlhughes replied:
removed.

adamvy reviewed (changes requested):
It looks like you have replaced the foam.u2.view.LoginView with foam.nanos.auth.login.LoginView.

But the new foam.nanos.auth.login.LoginView wasn't updated to include the oidc functionality added to the old foam.u2.view.LoginView, so this basically drops OIDC login support.

I think we need to delete the multiple LoginViews and signup/signin models, as it stands there's a bunch of code with overlapping responsibilties and its not clear which code is meant to be run.

But I also think we should add the oidcproviderdao code from the foam.u2.view.LoginView into foam.nanos.auth.login.LoginView so we can preserve oidc login support

adamvy commented on src/foam/nanos/auth/services.jrl:
```diff
@@ -275,6 +276,27 @@ p({
 
+p({
+  "class": "foam.nanos.boot.NSpec",
+  "name": "resetPasswordInsecureService",
```
I don't think we should enable/serve an "insecure" service by default.  That's a big footgun for anybody who builds on top of foam to forget to disable this service.

jlhughes replied:
This entire PR is a work in progress, so this is in place until we get everything working.  This is also mentioned in the PR notes.

jlhughes replied:
Have another  app sitting on foam to test application level things, so disabled now instead of just when we merge to dev.

jlhughes committed:
revert image path - not required

jlhughes committed:
remove unused import of memento

jlhughes committed:
Merge branch 'SignInSignUp' of https://github.com/kgrgreer/foam3 into SignInSignUp

jlhughes committed:
Merge branch 'development' of https://github.com/kgrgreer/foam3 into SignInSignUp

jlhughes committed:
Merge remote-tracking branch 'origin/SignInSignUp-updateloginview' into SignInSignUp

noodlemoodle committed:
update loginview (#3585)

* update loginview

* fix resetpassword visibility

* add signInWithOIDC to clientloginservice

* use loginservice signinwithoidc

* fix  actionview

* fix login on enter

jlhughes committed:
Merge branch 'SignInSignUp' of github.com:kgrgreer/foam3 into SignInSignUp

jlhughes committed:
move password management setup examples to the bottom and show them with an application group rather than basicUser

jlhughes committed:
remove documentation on lifecycle which was stated why it changed.  Also comment out validateObj on desiredPassword field as it's causing the edit view to not render propertly

jlhughes committed:
cleanup and document model

jlhughes committed:
disable resetPasswordInsecureService.  remove File decorators on userRegistrationDAO as not required and causing json parsing error

jlhughes committed:
call await this.clientPromise before this.requestLogin to ensure client is completely setup before sign-in or sign-up are setup, else they may fail with missing nspecs.

jlhughes committed:
remove ACTIVE restriction. The uniqueUserService nspec adds the condition LifecycleState != DELETED. Additionally since this condition is in the nspec it makes it easy for an application configuration to override

jlhughes committed:
Merge branch 'SignInSignUp' of github.com:kgrgreer/foam3 into SignInSignUp

jlhughes committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into SignInSignUp

noodlemoodle committed:
make oidc show up

jlhughes committed:
remove debugger line

jlhughes committed:
Remove nspec userUpdatePassword and ResetPasswordInsecureService

Reverted userDAO setup to version earlier than clientLoginService PR,
which works just fine for simple user change password, and also admin update

jlhughes committed:
replace ctrl.add(Notification with this.notify, and ctrl.groupLoadingHandled with this.ctrl.groupLoadingHandled as it was using the global ctrl rather than the imported

jlhughes committed:
default notify transient to true

noodlemoodle committed:
signinsignup cleanup

noodlemoodle commented:
(NEW) cleanup prs
https://github.com/kgrgreer/foam3/pull/3587
https://github.com/nanoPayinc/NANOPAY/pull/18258

adamvy commented on src/foam/nanos/auth/UserRegistrationDAO.js:
```diff
@@ -0,0 +1,116 @@
+          .where(
+            AND(
+              EQ(User.USER_NAME, user.getUserName()),
+              EQ(User.LIFECYCLE_STATE, LifecycleState.ACTIVE),
```
Should this be checking for NEQ(LIFECYCLE_STATE, DELETED) instead?

jlhughes replied:
Updated. And yes, this is inline with other Lifecycle work I'm doing and is now consistent with the other changes we've made in this PR.

jlhughes committed:
Merge branch 'SignInSignUp' of github.com:kgrgreer/foam3 into SignInSignUp

jlhughes committed:
for duplicate check,check for NEQ DELETED rather than ACTIVE as this is consistent with other changes which only allow duplicate for DELETED

jlhughes committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into SignInSignUp

jlhughes committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into SignInSignUp

jlhughes committed:
Merge remote-tracking branch 'origin/signinsignup-cleanup' into SignInSignUp

jlhughes committed:
add closing }) - presumably lost in a bad merge

jlhughes committed:
comment out toast notification of failure to fetch group - as this is expected in the requestLogin flow

jlhughes committed:
if subject user is system, don't use it's spid to find a theme, rather just use the theme of the domain

jlhughes committed:
tests.jrl path was incorrect for PreventDuplicateEmailTest

jlhughes committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into SignInSignUp

jlhughes committed:
Remove ResetPasswordInsecure as UpdatePassword already handles this case.

UpdatePassword already handles logged in password change - didn't see it
with all the other changes.
Remove PasswordHashingDAO from regular user dao stacks.  Only required on
registration daos.

jlhughes committed:
rename logedInUser to loggedInUser

jlhughes committed:
revert context change, completely wrong. Also make use of spid in context as already setup by session

jlhughes committed:
UserPasswordHashingDAO - dao is responsible for hashing a desired password. If a desiredPassword property was empty, the logic recreated the user from the old user - the reason and usefullness for this can no longer be determined, and has been removed

jlhughes committed:
userDAO add back UserPasswordHashingDAO - else no way for admin to update users password

jlhughes committed:
In ConfirmationDialog.addActions declare argument X/this, passed by caller and use this to access model properties. Access through 'this' stopped working

jlhughes committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into SignInSignUp

jlhughes committed:
anonymous user - remove password and set loginEnabled:false to reduce risk of user being compromised

jlhughes committed:
revert loginService serve:true

jlhughes committed:
Merge branch 'SignInSignUp' of github.com:kgrgreer/foam3 into SignInSignUp

jlhughes committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into SignInSignUp
